common/persistence/sql/sqlplugin/mysql/task.go

```diff
@@ -139,6 +143,9 @@ func (mdb *db) UpdateTaskQueues(row *sqlplugin.TaskQueuesRow) (sql.Result, error
 func (mdb *db) SelectFromTaskQueues(filter *sqlplugin.TaskQueuesFilter) ([]sqlplugin.TaskQueuesRow, error) {
 	switch {
 	case filter.NamespaceID != nil && filter.Name != nil && filter.TaskType != nil:
+		if filter.ShardIDLessThanEqualTo != 0 || filter.ShardIDGreaterThanEqualTo != 0 {
+			return nil, fmt.Errorf("shardID range not supported for specific selection")
```

Comment:
Can we define this as a service error?  Returning untyped errors throws of how we emit metric from persistence metric client.